Edit Developer Guide Glossary (#15)
AY2223S1-CS2103T-W11-3/tp: docs/DeveloperGuide.md

(For all use cases below, the **System** is the `AddressBook` and the **Actor** is the `user`, unless specified otherwise)

**Use case: View a Customer**

**MSS**

1.  User requests to list customers
2.  ArtBuddy shows a list of customers
3.  User requests to open customer in the list

> 4.  ArtBuddy shows the list of commissions the customer made and the detail of the customer

Reviewer: ```suggestion
4.  ArtBuddy shows the list of commissions the customer made and the details of the customer
```

* 3a. The given index is invalid.

    * 3a1. ArtBuddy shows an error message.

      Use case resumes at step 2.

**Use case: Delete a Customer**

**MSS**

1.  User requests to list customers
2.  ArtBuddy shows a list of customers
3.  User requests to delete a customer in the list

> 4.  ArtBuddy deletes the customer

Reviewer 2: ```suggestion
4.  ArtBuddy warns the user that all commissions under the customer will be deleted, and confirms whether the user still wants to proceed
5.  User confirms.
6.  ArtBuddy deletes the customer
```

    Use case ends.

**Extensions**

* 2a. The list is empty.

  Use case ends.

* 3a. The given index is invalid.

    * 3a1. ArtBuddy shows an error message.

      Use case resumes at step 2.

> 

Reviewer 2: ```suggestion

* 4a. The user chooses not to proceed with the deletion.

  Use case ends.

```

### Glossary

* **Mainstream OS**: Windows, Linux, Unix, OS-X
* **Customer**: A contact detail. Contains information about the customer, and a list of commissions.

> * **Commission**: An art piece requested by a customer that has been delivered or under progress. Contains specifics about the commission and list of iterations.

Reviewer: ```suggestion
* **Commission**: An art piece requested by a customer that has been delivered or is in progress. Contains specifics about the commission and a list of iterations.
```

> * **Customer**: A contact detail, contains information about the customer, and a list of Commissions.
> * **Commission**: A single commission made by a customer, contains specifics about the commission and list of Iterations.
> * **Iteration**: A single iteration of the commission, contains an image for the commission and a text comment on the image. 

Reviewer 2: ```suggestion
* **Customer**: A contact detail. Contains information about the customer, and a list of commissions.
* **Commission**: An art piece requested by a customer. Contains specifics about the commission and list of iterations.
* **Iteration**: A single version of a commission. Contains an image and a text comment on the image. 
```
